src/pipeline/src/etl/value.rs

```diff
@@ -331,3 +333,79 @@ impl TryFrom<&yaml_rust::Yaml> for Value {
         }
     }
 }
+
+impl<'a> From<&Value> for JsonbValue<'a> {
+    fn from(value: &Value) -> Self {
+        match value {
+            Value::Null => JsonbValue::Null,
+            Value::Boolean(v) => JsonbValue::Bool(*v),
+
+            Value::Int8(v) => JsonbValue::Number(JsonbNumber::Int64(*v as i64)),
+            Value::Int16(v) => JsonbValue::Number(JsonbNumber::Int64(*v as i64)),
+            Value::Int32(v) => JsonbValue::Number(JsonbNumber::Int64(*v as i64)),
+            Value::Int64(v) => JsonbValue::Number(JsonbNumber::Int64(*v)),
+
+            Value::Uint8(v) => JsonbValue::Number(JsonbNumber::UInt64(*v as u64)),
+            Value::Uint16(v) => JsonbValue::Number(JsonbNumber::UInt64(*v as u64)),
+            Value::Uint32(v) => JsonbValue::Number(JsonbNumber::UInt64(*v as u64)),
+            Value::Uint64(v) => JsonbValue::Number(JsonbNumber::UInt64(*v)),
+            Value::Float32(v) => JsonbValue::Number(JsonbNumber::Float64(*v as f64)),
+            Value::Float64(v) => JsonbValue::Number(JsonbNumber::Float64(*v)),
+            Value::String(v) => JsonbValue::String(v.clone().into()),
+            Value::Timestamp(v) => JsonbValue::String(v.to_string().into()),
+            Value::Array(arr) => {
+                let mut vals: Vec<JsonbValue> = Vec::with_capacity(arr.len());
+                for val in arr.iter() {
+                    vals.push(val.into());
+                }
+                JsonbValue::Array(vals)
+            }
+            Value::Map(obj) => {
+                let mut map = JsonbObject::new();
+                for (k, v) in obj.iter() {
+                    let val: JsonbValue = v.into();
+                    map.insert(k.to_string(), val);
+                }
+                JsonbValue::Object(map)
+            }
+        }
+    }
+}
+
+impl<'a> From<Value> for JsonbValue<'a> {
```

Comment:
`impl<'a> From<&Value> for JsonbValue<'a> {`
`impl<'a> From<Value> for JsonbValue<'a> {`

These two `From` methods look extremely similar. Can we merge them into one?

Reply:
There's no good way to avoid it here.